### PR TITLE
📖 Rewrite the .ts code docs

### PR DESCRIPTION
**Pull request by ElRaptorus**
**Changes:**

Change the code documentations of the .TS implementation to make it more comprehensive.
1. Add `@throws` sections to all API functions
2. `@returns` now only describes the values that the function returns, not the Promise that wraps the function. That is already covered by the `@async` tag.
3. Remove some docs that serve no useful purpose, such as `The Id.` or `The list.`.
4. Unify usages of terms such as `ID` or `FormFields`.

PR: #68

## PR-Checklist

Please check the boxes in this list after submitting your PR:

- [x] You can merge this PR **right now** (if not, please prefix the title with "WIP: ")
- [x] I've tested **all** changes included in this PR.
- [x] I've also reviewed this PR myself before submitting (e.g. for scrambled letters, typos, etc.).
- [x] I've rebased the `develop` branch with my branch before finishing this PR.
- [x] I've **summarized all changes** in a list above.
- [x] I've mentioned all **PRs, which relate to this one**.
- [x] I've prefixed my Pull Request title is according to [gitmoji guide](https://gitmoji.carloscuesta.me/).

### CODE REVIEW

File: typescript/src/apis/iempty_activity_consumer_api.ts [hunk 1]
```diff
@@ -44,13 +50,16 @@ export interface IEmptyActivityConsumerApi {
...
-   *                      EmptyActivities.
-   * @returns             A Promise, which resolves with the retrieved EmptyActivities,
-   *                      or rejects an error, in case the request failed.
-   *                      This can happen, if the Correlation was not found,
-   *                      or the user is not authorized to see it.
+   * @param   identity           The requesting users identity.
+   * @param   correlationId      The ID of the Correlation for which to
+   *                             retrieve the EmptyActivities.
+   * @returns                    A list of waiting EmptActivities for the given
+   *                             Correlation.
```
Comment (cmg-dev) [inline typescript/src/apis/iempty_activity_consumer_api.ts]:
```suggestion
   *                             Correlation.
```

is correlation a name? if not -> correlation.

**Comment by ElRaptorus** [comment]
@5ebastianMeier Because they serve no useful purpose. I mentioned as much in the PR body.

File: typescript/src/apis/iempty_activity_consumer_api.ts (hunk 1 above)
Comment (ElRaptorus) [reply typescript/src/apis/iempty_activity_consumer_api.ts]: yes it is.

**Comment by 5ebastianMeier** [comment]
@ElRaptorus I know those docs can be a chore and often just express the variable name in a short sentence.

But in terms of overall quality I prefer to decide for or against the docs and stick with it - even in these cases.

Also we already had the documentation for these so it wouldn't hurt to just leave it there. There is also no extra effort to maintain them as such docs are unlikely to change.

If we decide to generate a reference for our codebase (the former `gulp doc` task already did that per repo), there's holes in that and it becomes inconsistent.

**Comment by ElRaptorus** [comment]
@5ebastianMeier **IF** we ever do that, we'll likely have to overhaul all the contract docs anyway, because pretty much all of the contracts are already documented in the way that I have adjusted the Consumer API contracts to. 
But since I don't think it likely that this is going to happen any time soon (if ever), I'd like to keep the style consistent across all the APIs.

And, no offence, but even **IF** we should decide to generate such docs, I'd rather not fall back to that infernal gulp task, given the trouble it has given us in the past, just by existing. 😉

**Comment by cmg-dev** [comment]
@ElRaptorus 

I think the removal of these docs is a good thing.

Don't want to go too deep into this discussion Maelstrom, but in ancient times we agreed on giving useful documentation. 

And if an 

```ts
export class ProcessModelList {
  /**
   * The ProcessModel list.
   */
  public processModels: Array<ProcessModel> = [];
}
```

where three lines are expressing the very same thing, we can probably agree on, that this is not very useful.

**Pull request merged**